src/file_import_retriever.rs

```diff
@@ -0,0 +1,162 @@
+use std::collections::{HashMap, HashSet};
+use std::iter::FromIterator;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use swc_common::errors::{ColorConfig, Handler};
+use swc_common::source_map::Pos;
+use swc_ecma_ast::Str;
+use swc_ecma_parser::Capturing;
+use swc_ecma_parser::{lexer::Lexer, Parser, StringInput, Syntax};
+pub struct SourceSpecifiers {
+    specifiers: Vec<swc_ecma_ast::ImportSpecifier>,
+    source: Str,
+}
+
+pub fn get_imports_from_file<'a>(file_path: &'a PathBuf) -> Vec<SourceSpecifiers> {
+    let cm = Arc::<swc_common::SourceMap>::default();
+    let fm = cm
+        .load_file(Path::new(file_path.to_str().unwrap()))
+        .expect("Could not load file");
+    let handler = Handler::with_tty_emitter(ColorConfig::Auto, true, false, Some(cm.clone()));
+
+    let lexer = create_lexer(&fm);
+
+    let capturing = Capturing::new(lexer);
+
+    let mut parser = Parser::new_from(capturing);
+
+    for e in parser.take_errors() {
+        e.into_diagnostic(&handler).emit();
+    }
+
+    let ts_module = parser
+        .parse_typescript_module()
+        .map_err(|e| e.into_diagnostic(&handler).emit())
+        .expect("Failed to parse module.");
+
+    let imports: Vec<_> = ts_module
+        .body
+        .iter()
+        .filter_map(|node| -> Option<SourceSpecifiers> {
+            if node.is_module_decl() {
+                if let Some(module_decl) = node.as_module_decl() {
+                    if module_decl.is_import() {
+                        let i = node.as_module_decl().unwrap().as_import().unwrap();
+                        return Some(SourceSpecifiers {
+                            specifiers: i.specifiers.clone().to_vec(),
+                            source: i.src.clone(),
+                        });
+                    }
+                }
+            }
+            None
+        })
+        .collect();
+
+    return imports;
+}
+
+pub fn get_string_of_span<'a>(file_text: &'a Vec<u8>, span: &'a swc_common::Span) -> String {
+    String::from_utf8_lossy(&file_text[span.lo().to_usize() - 1..span.hi().to_usize() - 1])
+        .to_string()
+}
+
+pub fn get_imports_map(
+    imports: &Vec<SourceSpecifiers>,
+    importer_file_path: &PathBuf,
+) -> HashMap<String, Option<HashSet<String>>> {
+    let mut imports_map: HashMap<String, Option<HashSet<String>>> = HashMap::new();
+
+    imports.iter().for_each(|import| {
+        let set: HashSet<String> = import
+					.specifiers
+					.iter()
+					.filter_map(|spec| -> Option<String> {
+						let file_text =std::fs::read(importer_file_path).expect(&format!("error opening source file \"{:?}\"", importer_file_path));
```

Comment:
can you run `cargo fmt` before merging? just to clean up this weird whitespace. If it gets messy / has unrelated changes it could be a follow-up PR.